### PR TITLE
fix: onboard custom alias support for braze

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

The following changes introduces feasibility of customising alias details for an anonymous user to braze. Custim alias label and alias id can be passed via integrations object and same would be used for aliasing the unidentified user to identified user

## What is the related Linear task?

Resolves INT-2091

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

File: src/v0/destinations/braze/util.js [hunk 1]
```diff
@@ -518,8 +518,18 @@ function setExternalId(payload, message) {
   return payload;
 }
 
-function setAliasObjectWithAnonId(payload, message) {
-  if (message.anonymousId) {
+function setAliasObject(payload, message) {
+  const integrationsObj = getIntegrationsObj(message, 'BRAZE');
```
Comment (karlrl) [inline src/v0/destinations/braze/util.js:522]: suggestion: does this need to be all caps? There's a mix of cases for object name, but lowercase seems more common.

```suggestion
  const integrationsObj = getIntegrationsObj(message, 'braze');
```

**Review by karlrl — COMMENTED** [review]
I didn't notice tests that included a `userId` but otherwise this looks good to me.

This change would come in handy for some work I'm doing. Thanks for the PR!

File: src/v0/destinations/braze/util.js (hunk 1 above)
Comment (utsabc) [reply src/v0/destinations/braze/util.js:522]: We use canonical names to resolve mix of casing for integration Object inputs  [ref1](https://github.com/rudderlabs/rudder-transformer/blob/673311f418b387aa17b6a2e0de6b6dce2871c672/src/constants/destinationCanonicalNames.js#L143) , 
[ref2](https://github.com/rudderlabs/rudder-transformer/blob/673311f418b387aa17b6a2e0de6b6dce2871c672/src/v0/util/index.js#L1005)

**Comment by utsabc** [comment]
> I didn't notice tests that included a `userId` but otherwise this looks good to me.
> 
> This change would come in handy for some work I'm doing. Thanks for the PR!

Sure will update

File: src/v0/destinations/braze/util.js (hunk 1 above)
Comment (karlrl) [reply src/v0/destinations/braze/util.js:522]: Ah ok! Thanks for the extra context.

File: src/util/prometheus.js [hunk 2]
```diff
@@ -607,6 +607,12 @@ class Prometheus {
         type: 'counter',
         labelNames: ['destination_id'],
       },
+      {
+        name: 'braze_alias_missConfigured_count',
```
[the remark this answers is not in the document]
Comment (utsabc) [reply src/util/prometheus.js]: Fixed

**Review by koladilip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

**Pull request merged**